### PR TITLE
[core,exec,lib] fix effective config write path with direct-runner

### PR DESCRIPTION
**Pull request by DanSimon**
This fixes a slight issue when using direct-runner.

Because `_load_config_from_file` is looking for the runtime config at `/usr/local/klio-job-run-effective.yaml`, we need to make sure that `klioexec` writes the runtime config directly to this path when using direct-runner, instead of writing it to `/usr/src/app/...` to be included in the distribution package for dataflow.

## Checklist for PR author(s)

- [ ] Format the pull request title like `[cli] Fixes bugs in 'klio job fake-cmd'`.
- [ ] Changes are covered by unit tests (no major decrease in code coverage %) and/or integration tests.
- [ ] Document any relevant additions/changes in the appropriate spot in `docs/src`.
- [ ] For any change that affects users, update the package's changelog in ``docs/src/reference/<package>/changelog.rst``.

### CODE REVIEW

**Review by fallonchen — COMMENTED** [review]
Just one question!

File: exec/src/klio_exec/commands/run.py [hunk 1]
```diff
@@ -272,7 +272,10 @@ def _get_run_callable(self):
 
     def _write_run_effective_config(self):
         # this method assumes setup.py is being used!
-        path = config_core.RUN_EFFECTIVE_CONFIG_PATH
+        if self.runtime_conf.direct_runner:
```
[the remark this answers is not in the document]
Comment (econchick) [reply exec/src/klio_exec/commands/run.py:275]: @fallonchen it looks like this method is only called if the [setup.py file exists](https://github.com/spotify/klio/blob/release/exec/src/klio_exec/commands/run.py#L339) - does that address your concern/question?

**Review by econchick — APPROVED** [review]
lgtm! but just double check on @fallonchen 's question

File: exec/src/klio_exec/commands/run.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanSimon) [reply exec/src/klio_exec/commands/run.py:275]: Yeah to clarify, this method is only called in `_verify_packaging` which does check to see if `setup.py` is being used.

**Pull request merged**